foundry.md

```diff
@@ -458,12 +466,75 @@ This rule then takes from the function call data the account using `#asWord(#ran
       requires SELECTOR ==Int selector ( "store(address,bytes32,bytes32)" )
 ```
 
+#### expectRevert - expect the next call to revert.
+
+```
+function expectRevert() external;
+function expectRevert(bytes4 msg) external;
+function expectRevert(bytes calldata msg) external;
+```
+
+Ignore all cheat code calls which take place while `expectRevert` is active.
+
+```{.k .bytes}
+    rule [foundry.call.ignoreCalls]:
+         <k> #call_foundry _ _ => . ... </k>
+         <expected>
+           <expectedRevert> true </expectedRevert>
+           ...
+         </expected>
+      [priority(35)]
+```
+
+Catch reverts.
+If the current call depth is equal with the `expectedDepth` and the `expectedBytes` match the `<output>` cell, then replace the `EVMC_REVERT` status code with `EVMC_SUCCESS`.
+
+```{.k .bytes}
+    rule <statusCode> EVMC_REVERT => EVMC_SUCCESS </statusCode>
+         <k> #halt ~> #return _RETSTART _RETWIDTH ... </k>
+         <output> OUT </output>
+         <callDepth> CD </callDepth>
+         <expected>
+           <expectedRevert> true => false </expectedRevert>
+           <expectedDepth> CD </expectedDepth>
+           <expectedBytes> EXPECTED </expectedBytes>
+         </expected>
+      requires (EXPECTED =/=K .ByteArray andBool EXPECTED ==K #range(OUT, 4, #sizeByteArray(OUT) -Int 4))
+        orBool EXPECTED ==K .ByteArray
+      [priority(40)]
+```
+
+Change the status code from `EVMC_SUCCESS` to `EVMC_REVERT` if a revert is expected but the call succeded.
+
+```{.k .bytes}
+    rule <statusCode> EVMC_SUCCESS => EVMC_REVERT </statusCode>
+         <k> #halt ~> #return _RETSTART _RETWIDTH ... </k>
+         <callDepth> CD </callDepth>
+         <expected>
+           <expectedRevert> true => false </expectedRevert>
```

Comment:
So, if I'm reading this correctly, `expectRevert` is only working for 1 top-level revert. It won't work for nested reverts, or for multiple reverts. Is that correct?

Can we make a test case for a nested revert and a multi-revert (two calls that both revert, and the expectRevert should absorb the first one, but the second one causes test failure with a different message). We should test this behavior.

Reply:
I've added the requested tests.